commands/admin.js

```diff
@@ -0,0 +1,137 @@
+                });
+            }
+
+            if (interaction.options.getSubcommand() === COMMAND_KICKUNVERIFIED) {
+                const role = await interaction.guild.roles.cache.find(
+                    (r) => r.name.toLowerCase() === "unverified",
+                );
+
+                // Make sure that the "unverified" role exists
+                if (role === undefined) {
+                    return await interaction.reply('Error: no "unverified" role exists.');
+                }
+
+                const kickMessage =
+                    "You have been removed from the CSESoc Server as you have not verified via the instructions in #welcome.\
+                    If you wish to rejoin, visit https://cseso.cc/discord";
+
+                // Member list in the role is cached
+                let numRemoved = 0;
+                await role.members.each((member) => {
+                    member.createDM().then((DMChannel) => {
+                        // Send direct message to user being kicked
+                        DMChannel.send(kickMessage).then(() => {
+                            // Message sent, time to kick.
+                            member
+                                .kick(kickMessage)
+                                .then(() => {
+                                    ++numRemoved;
+                                    console.log(numRemoved + " people removed.");
+                                })
+                                .catch((e) => {
+                                    console.log(e);
+                                });
+                        });
+                    });
+                });
+                return await interaction.reply("Removed unverified members.");
+            } else if (interaction.options.getSubcommand() === COMMAND_MIGRATE) {
+                // get all roles, and find courses which match the regex
+                const course_roles = await interaction.guild.roles.cache.filter((role) =>
+                    is_valid_course(role.name),
+                );
+
+                // for each course role, find the corresponding channel
+                // and add a permission overwrite for each member with that role
+                // to that channel so that they can view it
+
+                for (const role of course_roles.values()) {
+                    const channel = await interaction.guild.channels.cache.find(
+                        (role_channel) =>
+                            role_channel.name.toLowerCase() === role.name.toLowerCase(),
+                    );
+
+                    if (channel === undefined) {
+                        console.log("No channel found for role " + role.name);
+                        continue;
+                    }
+
+                    for (const member of role.members.values()) {
+                        await channel.permissionOverwrites.create(member, {
+                            VIEW_CHANNEL: true,
```

Comment:
Was wondering if we need to give send message permission for these channels unless they are already given to everyone previously.

Reply:
this would be inherited from the @.everyone overwrite, which is send by default for anyone that has read permissions.